meshpy/mesh_creation_functions/nurbs_geometries.py

```diff
@@ -218,6 +218,125 @@ def create_nurbs_flat_plate_2d(width, length, *, n_ele_u=1, n_ele_v=1):
     return surf
 
 
+def create_nurbs_base_sphere_patch_surf(radius, n_ele_u=1, n_ele_v=1):
+    """
+    This function does a base patch to create a sphere.
+    This patch represents a surface.
+
+    Args
+    ---
+    radius: double
+        radius of the sphere
+    n_ele_u: int
+        number of elements in the parametric u-direction
+    n_ele_v: int
+        number of elements in the parametric v-direction
+
+    Return
+    ----
+    surf: geomdl object
+        geomdl object that contains the surface information
+    """
+
+    # Create a NURBS surface instance
```

Comment:
As far as I can tell this function creates a patch with the segment angle 90 and 90, i.e., 6 patches are needed to build a full sphere, right?

Is it also possible to let the user specify the angle?

Dis you come up with this implementation or is this taken from some literature, forum, ...? If so, please add a short reference.

Reply:
Exactly, each segment has an angle 90 and 90 and with 6 patches a full sphere is built. I took the implementation from the code of the `NURBS-generator` so that I can add it as a reference. For now, the user can't specify the angle, as I'm not sure how to determine the weights of the control points depending on the angle.